av1an-core/src/settings.rs

```diff
@@ -526,6 +529,16 @@ properly into a mkv file. Specify mkvmerge as the concatenation method by settin
       self.video_params = self.encoder.get_default_arguments();
     }
 
+    if let Some(grain) = self.grain_synth {
+      if grain > 64 {
+        bail!("Valid strength values for grain synth are 0-64");
+      }
+    }
+
+    if self.grain_synth.is_some() && self.encoder != Encoder::aom {
```

Comment:
We can probably move this check to be inside if the `if let Some(grain)` statement